lib/property_sets/active_record_extension.rb

```diff
@@ -55,10 +52,7 @@ def property_set(association, options = {}, &block)
         # stolen/adapted from AR's collection_association.rb #define_extensions
 
         module_name = "#{association.to_s.camelize}AssociationExtension"
-        module_name = name.demodulize + module_name unless RAILS6
-
-        target = RAILS6 ? self : self.parent
-        association_module = target.const_get module_name
+        association_module = self.const_get module_name
```

Comment:
Let’s fix lines 49-50 too, e.g.
```
        # eg: Account::SettingsAssociationExtension
```

Reply:
Do you mean removing these comments?

Reply:
I thought about keeping the 1 relevant comment, but we can remove them both 👍🏼